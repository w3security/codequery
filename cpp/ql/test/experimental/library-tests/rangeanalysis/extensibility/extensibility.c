/// Adds its arguments (has custom modeling in QL)
int custom_add_function(int a, int b);

int test_extensibility_add(int x) {
  if (x >= -10 && x <= 10) {
    int result = custom_add_function(x, 100);
    return result; // 90 .. 110
  }
}
