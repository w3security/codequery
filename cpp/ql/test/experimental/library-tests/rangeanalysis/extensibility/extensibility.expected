| extensibility.c:5:7:5:7 | x | -2.147483648E9 | 2.147483647E9 |
| extensibility.c:5:19:5:19 | x | -10.0 | 2.147483647E9 |
| extensibility.c:6:38:6:38 | x | -10.0 | 10.0 |
| extensibility.c:7:12:7:17 | result | 90.0 | 110.0 |
